loopd/daemon.go

```diff
@@ -161,75 +179,184 @@ func daemon(config *config, lisCfg *listenerCfg) error {
 		log.Infof("REST proxy disabled")
 	}
 
-	mainCtx, cancel := context.WithCancel(context.Background())
-	var wg sync.WaitGroup
+	// Start the main event handlers.
+	d.startEventHandlers()
 
-	// Start the swap client itself.
-	wg.Add(1)
+	// Start the grpc server.
+	d.wg.Add(1)
 	go func() {
-		defer wg.Done()
+		defer d.wg.Done()
 
-		log.Infof("Starting swap client")
-		err := swapClient.Run(mainCtx, server.statusChan)
+		log.Infof("RPC server listening on %s", d.grpcListener.Addr())
+		err = d.grpcServer.Serve(d.grpcListener)
 		if err != nil {
 			log.Error(err)
 		}
-		log.Infof("Swap client stopped")
+	}()
 
-		log.Infof("Stopping gRPC server")
-		grpcServer.Stop()
+	return nil
+}
 
-		cancel()
-	}()
+// StartAsSubserver is an alternative to Start where the RPC server does not
+// create its own gRPC server but registers to an existing one. The same goes
+// for REST (if enabled), instead of creating an own mux and HTTP server, we
+// register to an existing one.
+func (d *Daemon) StartAsSubserver(lndGrpc *lndclient.GrpcLndServices) error {
```

Comment:
Can this be called by the stand-alone `Start`?

Reply:
No, because we want to have the guard against starting it twice in there. I refactored some things so this got even smaller. I don't think we gain any readability by refactoring this into smaller methods.